packages/ember-data/lib/system/model/model.js

```diff
@@ -844,6 +817,35 @@ var Model = Ember.Object.extend(Ember.Evented, {
   },
 
   /**
+    @method _changedKeys
+    @private
+  */
+  _changedKeys: function(updates) {
+    var changedKeys = [];
+
+    if (updates && typeof updates === 'object') {
+      var original, i, value, key;
+      var keys = Ember.keys(updates);
+      var length = keys.length;
+
+      original = merge({}, this._data);
+      original = merge(original, this._attributes);
```

Comment:
However, if the server were to return something completely different, say `{ name: 'Igor' }`, we would get a trigger, but the value would remain `{ name: 'Tom' }`. That might be an issue though.